git/libgit2/repository_client.go

```diff
@@ -248,22 +248,18 @@ func (l *Client) Commit(info git.Commit, signer *openpgp.Entity) (string, error)
 	}
 	defer index.Free()
 
-	// add to index any files that are not within .git/
-	if err = filepath.Walk(l.repository.Workdir(),
+	if err = util.Walk(l.repoFS, "",
 		func(path string, info os.FileInfo, err error) error {
 			if err != nil {
 				return err
 			}
-
-			rel, err := filepath.Rel(l.repository.Workdir(), path)
-			if err != nil {
-				return err
-			}
-			if info.IsDir() || strings.HasPrefix(rel, ".git") || rel == "." {
+			// skip symlinks and any files that are within ".git/"
+			if !info.Mode().IsRegular() || info.IsDir() ||
+				strings.HasPrefix(path, ".git") || path == "" {
```

Comment:
We should be able to drop the `info.IsDir()` due to https://github.com/golang/go/blob/fa0e3bffb494fb9f97c241d6579daa443fbcbb06/src/os/types.go#L38

```suggestion
			if !info.Mode().IsRegular() || 		strings.HasPrefix(path, ".git") || path == "" {
```